src/envs/behavior.py

```diff
@@ -453,6 +458,13 @@ def _reachable_classifier(self, state: State,
         assert len(objs) == 2
         ig_obj = self.object_to_ig_object(objs[0])
         ig_other_obj = self.object_to_ig_object(objs[1])
+        # If the two objects are the same (i.e reachable(agent, agent)),
+        # we always want to return False so that when we learn
+        # operators, such predicates don't needlessly appear in preconditions.
+        # In the future, we probably want to prevent such atoms from
```

Comment:
I don't get it -- why is this in the future? Won't this change prevent such atoms from ever getting created?

Reply:
i think nishanth means that we should never even call predicate classifiers with repeated arguments. but idk about that, you could totally imagine some settings where repeated arguments make sense. i think it's best to leave it to the classifier to decide what it wants to do wrt repeated argument, so like you're doing now already. i'm in favor of removing this sentence "In the future, ..."